File: library/lib_module_http/src/main/java/org/cchao/http/DeleteCacheConsumer.java
package org.cchao.http;

import org.cchao.common.utils.JsonUtils;
import org.cchao.common.utils.Md5Utils;
import org.cchao.http.db.CacheDbUtils;
import org.cchao.http.db.CacheModel;

import io.reactivex.functions.Consumer;

class DeleteCacheConsumer implements Consumer<Throwable> {

    private HttpRequestBody httpRequestBody;

    private boolean isCache;

    DeleteCacheConsumer(HttpRequestBody httpRequestBody, boolean isCache) {
        this.httpRequestBody = httpRequestBody;
        this.isCache = isCache;
    }

    @Override
    public void accept(Throwable throwable) throws Exception {
        if (isCache) {
            String key = Md5Utils.getMd5(JsonUtils.toString(httpRequestBody));
            CacheModel cacheModel = CacheDbUtils.getInstance().queryCacheModel(key);
            if (null != cacheModel) {
                CacheDbUtils.getInstance().deleteCache(cacheModel);
            }
        }
    }
}

File: library/lib_module_http/src/main/java/org/cchao/http/DeleteCacheConsumer.kt
package org.cchao.http

import io.reactivex.rxjava3.functions.Consumer
import org.cchao.common.utils.JsonUtils
import org.cchao.common.utils.Md5Utils
import org.cchao.http.db.CacheDbUtils

internal class DeleteCacheConsumer(private val httpRequestBody: HttpRequestBody, private val isCache: Boolean) : Consumer<Throwable> {

    override fun accept(throwable: Throwable) {
        if (isCache) {
            val key = Md5Utils.getMd5(JsonUtils.toString(httpRequestBody)).plus(httpRequestBody.url)
            val cacheModel = CacheDbUtils.instance.queryCacheModel(key)
            if (null != cacheModel) {
                CacheDbUtils.instance.deleteCache(cacheModel)
            }
        }
    }
}
